test/wait.go

```diff
@@ -147,6 +167,11 @@ func WaitForServiceExternalIPState(ctx context.Context, c *clients, namespace, n
 	defer span.End()
 
 	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		select {
```

Comment:
I'm worried future callers will forget this stanza at the top of their funcs. Can we somehow make this the responsibility of PollImmediate so it's not the caller's responsibility?

Reply:
I think I have another shape in mind that you might be happier with, gimme a few minutes 😉

Reply:
It ends up being gross too, lemme write a shared helper...

Reply:
alright, I think I pushed a change that better centralizes the context cancellation handling logic